stainless_extraction/src/expr/mod.rs

```diff
@@ -349,6 +356,17 @@ impl<'a, 'l, 'tcx> BodyExtractor<'a, 'l, 'tcx> {
     }
   }
 
+  /// Inserts a fresh copy around the expression, if the expression can contain
+  /// mutable types. This function must only be called, when it is safe to add a
+  /// fresh copy.
```

Comment:
Currently, this is safe everywhere because:
- we either work on a immutable reference aka shared borrow aka `& T`
- or we _own_ the the value, meaning it's moved/consumed, when we return/alias/shadow it somewhere and Rust has already made sure that it can't be used afterwards.

Reply:
I would adapt the comment to reflect what you said in the comments above. Otherwise, as the reader, I'd get an uneasy feeling of not understanding how to safely use this function.